glib/src/value.rs

```diff
@@ -904,6 +904,49 @@ impl ToValueOptional for String {
     }
 }
 
+impl ValueType for Box<str> {
+    type Type = String;
+}
+
+impl ValueTypeOptional for Box<str> {}
+
+unsafe impl<'a> FromValue<'a> for Box<str> {
+    type Checker = GenericValueTypeOrNoneChecker<Self>;
+
+    unsafe fn from_value(value: &'a Value) -> Self {
+        Box::<str>::from(<&str>::from_value(value))
+    }
+}
+
+impl StaticType for Box<str> {
+    fn static_type() -> Type {
+        String::static_type()
+    }
+}
+
+impl ToValue for Box<str> {
+    fn to_value(&self) -> Value {
+        <&str>::to_value(&self.as_ref())
+    }
+
+    fn value_type(&self) -> Type {
+        String::static_type()
```

Comment:
Why not self.value_type()?

Reply:
I simply copied it from `String` implementation of `value_type`, and it seems that the implementation of `str` of `value_type` also calls into `String::static_type()`.